internal/cmd/kafka/command_link_create.go

```diff
@@ -206,44 +320,53 @@ func (c *linkCommand) addSecurityConfigToMap(cmd *cobra.Command, linkMode linkMo
 		return err
 	}
 	if sourceApiKey != "" && sourceApiSecret != "" {
-		if linkMode == Destination {
-			// For links at destination cluster, the credentials are for the remote cluster.
-			configMap[securityProtocolPropertyName] = saslSsl
-			configMap[saslMechanismPropertyName] = plain
-			configMap[saslJaasConfigPropertyName] = getJaasValue(sourceApiKey, sourceApiSecret)
-		} else {
-			// For source initiated links at source cluster, the credentials are for the local cluster.
-			configMap[localListenerPropertyName] = saslSsl
-			configMap[localSecurityProtocolPropertyName] = saslSsl
-			configMap[localSaslMechanismPropertyName] = plain
-			configMap[localSaslJaasConfigPropertyName] = getJaasValue(sourceApiKey, sourceApiSecret)
-		}
+		configMap[securityProtocolPropertyName] = saslSsl
+		configMap[saslMechanismPropertyName] = plain
+		configMap[saslJaasConfigPropertyName] = getJaasValue(sourceApiKey, sourceApiSecret)
 	} else if sourceApiKey != "" || sourceApiSecret != "" {
 		return errors.New("--source-api-key and --source-api-secret must be supplied together")
 	}
+	return nil
+}
 
-	if linkMode == Source {
-		destinationApiKey, err := cmd.Flags().GetString(destinationApiKeyFlagName)
-		if err != nil {
-			return err
-		}
-		destinationApiSecret, err := cmd.Flags().GetString(destinationApiSecretFlagName)
-		if err != nil {
-			return err
-		}
-		if destinationApiKey != "" && destinationApiSecret != "" {
-			// For source initiated links at source cluster, the credentials are for the remote cluster.
-			configMap[securityProtocolPropertyName] = saslSsl
-			configMap[saslMechanismPropertyName] = plain
-			configMap[saslJaasConfigPropertyName] = getJaasValue(destinationApiKey, destinationApiSecret)
-		} else if destinationApiKey != "" || destinationApiSecret != "" {
-			return errors.New("--destination-api-key and --destination-api-secret must be supplied together")
-		}
+func (c *linkCommand) addSourceInitiatedLinkSecurityConfigToMap(cmd *cobra.Command, configMap map[string]string) error {
+	sourceApiKey, err := cmd.Flags().GetString(sourceApiKeyFlagName)
+	if err != nil {
+		return err
+	}
+	sourceApiSecret, err := cmd.Flags().GetString(sourceApiSecretFlagName)
+	if err != nil {
+		return err
+	}
+	if sourceApiKey != "" && sourceApiSecret != "" {
+		configMap[localListenerPropertyName] = saslSsl
+		configMap[localSecurityProtocolPropertyName] = saslSsl
+		configMap[localSaslMechanismPropertyName] = plain
+		configMap[localSaslJaasConfigPropertyName] = getJaasValue(sourceApiKey, sourceApiSecret)
+	} else if sourceApiKey != "" || sourceApiSecret != "" {
+		return errors.New("--source-api-key and --source-api-secret must be supplied together")
```

Comment:
You can use the built-in function, `cmd.MarkFlagsRequiredTogether("a", "b")`

Reply:
Hm are you sure? GoLand doesnt like that method and it also doesnt compile. I get:
```
  ⨯ build failed after 2s                    error=failed to build for darwin_amd64_v1: exit status 1: # github.com/confluentinc/cli/internal/cmd/kafka
internal/cmd/kafka/command_link_create.go:108:14: cmd.Flags().MarkFlagsRequiredTogether undefined (type *pflag.FlagSet has no field or method MarkFlagsRequiredTogether)

make[1]: *** [cli-builder] Error 1
make: *** [build] Error 2
```

Reply:
Instead of `cmd.Flags().MarkFlagsRequiredTogether()` try `cmd.MarkFlagsRequiredTogether()`

Reply:
(There should be at least one other example of it being used in the CLI, if you want to search for it)

Reply:
Oh never mind I found it.